src/validation.h

```diff
@@ -366,13 +366,19 @@ static_assert(std::is_nothrow_destructible_v<CScriptCheck>);
  */
 class ValidationCache
 {
+private:
+    //! Pre-initialized hasher to avoid having to recreate it for every hash calculation.
+    CSHA256 m_script_execution_cache_hasher;
+
 public:
     CuckooCache::cache<uint256, SignatureCacheHasher> m_script_execution_cache;
 
     ValidationCache(size_t script_execution_cache_bytes);
 
     ValidationCache(const ValidationCache&) = delete;
     ValidationCache& operator=(const ValidationCache&) = delete;
+
+    CSHA256 ScriptExecutionCacheHasher() const { return m_script_execution_cache_hasher; }
```

Comment:
maybe a comment to document that we intend to make a copy of the pre-initialized hasher?